client/src/components/jobPosting/TimespanPicker.js

```diff
@@ -19,8 +19,29 @@ export class TimespanPicker extends Component {
   }
 
   componentDidMount() {
-    this.props.addShowTo(null)
-    this.props.addShowFrom(null)
+    if (this.props.setTimespan) {
+      const now = moment()
+      const copiedShowFrom = moment(this.props.showFrom)
```

Comment:
This statement needs a check `&& this.props.showFrom`, because `moment(null) = NaN`, which causes text 'Invalid date' in the input field.